kytos/core/api_server.py

```diff
@@ -130,3 +134,65 @@ def shutdown_api(self):
     def web_ui(self):
         """Method used to serve the index.html page for the admin-ui."""
         return send_from_directory(self.flask_dir, 'index.html')
+
+    # BEGIN decorator methods
+
+    @staticmethod
+    def decorator(rule, **options):
+        """Decorator for REST endpoints using Flask.
+
+        Parameter documentation is the same as Flask's @app.route. As we don't
+        have the NApp instance now, we store the parameters in a method
+        attribute in order to add the route later, after APIServer is the NApp
+        object is passed as a parameter of :meth:`start_rest_api`.
+        """
+        def store_route_params(function):
+            """Store ``Flask`` ``@route`` parameters in a method attribute."""
+            function.route_params = rule, options
+            return function
+        return store_route_params
+
+    def start_rest_api(self, napp):
+        """Start ``napp`` REST endpoints."""
+        for method in self._get_api_methods(napp):
+            rule, options = method.route_params
+            self._start_endpoint(rule, method, **options)
+
+    @staticmethod
+    def _get_api_methods(napp):
+        """Return ``napp``'s methods having the @rest decorator."""
+        for name in dir(napp):
+            if not name.startswith('_'):  # discarding private names
+                pub_attr = getattr(napp, name)
+                if callable(pub_attr) and hasattr(pub_attr, 'route_params'):
+                    yield pub_attr
+
+    def _start_endpoint(self, rule, function, **options):
+        """Start ``function``'s endpoint.
+
+        1. Add '/kytos' prefix to the URL;
+        2. Warn if the same URL and method were already started.
+        2. Forward parameters to ``Flask.add_url_rule`` mimicking
+           Flask ``@route`` decorator.
+        """
+        rule = f'/kytos{rule}'
```

Comment:
Let's try to add `api/<author>/<napp_name>/` as prefix.

Reply:
This is strange...